R/MsBackendDataFrame.R

```diff
@@ -97,21 +95,20 @@ setMethod("as.list", "MsBackendDataFrame", function(x) {
 
 #' @rdname hidden_aliases
 setMethod("centroided", "MsBackendDataFrame", function(object) {
-    .get_rle_column(object@spectraData, "centroided")
+    .get_column(object@spectraData, "centroided")
 })
 
 #' @rdname hidden_aliases
 #'
 #' @aliases centroided<-,MsBackendDataFrame-method
 #'
-#' @importFrom MsCoreUtils asRle
 setReplaceMethod("centroided", "MsBackendDataFrame", function(object, value) {
     value_len <- length(value)
     value_type <- is.logical(value)
     if (value_len == 1 && value_type)
-        object@spectraData$centroided <- Rle(value, length(object))
+        object@spectraData$centroided <- rep_len(value, length(object))
     else if (value_len == length(object) && value_type)
-        object@spectraData$centroided <- asRle(value)
+        object@spectraData$centroided <- value
```

Comment:
Do we really need this `if`/`else` here? `data.frame(...)$column <- value` will replicate the value automatically if `length(value) == 1`.

Reply:
The only reason that I left the `if` `else` here is that I wanted to avoid the automatic replication of R (i.e. if the `DataFrame` has 12 rows, `value` can also be of length 2, 3, 4 or 6 - but I wanted only to support either length 1 or length equal nrow of `DataFrame`).

Reply:
Ok, but then something like the following should be enough, or?
```r
if (value_type && (value_len == 1L || value_len == length(object)))
    object@spectraData$centroided <- value  
```

Reply:
OK. I will adapt.

Reply:
Changed now